src/poetry/publishing/uploader.py

```diff
@@ -275,7 +275,9 @@ def _upload_file(
                         "Redirects are not supported. "
                         "Is the URL missing a trailing slash?"
                     )
-                elif resp.status_code == 400 and "was ever registered" in resp.text:
+                elif (
```

Comment:
this code branch looks as though it was aimed at the specific behaviour of a particular server (almost surely pypi).  I doubt that broadening the range of status codes is useful